### PR TITLE
#3378 Minor refactor for Shabondi sink

### PR DESCRIPTION
**Pull request by vitojeng**
required by #3378 

### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #issue_number)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala [hunk 1]
```diff
@@ -76,15 +62,15 @@ private[shabondi] class RowQueueProducer(
...
+            val rows: Seq[Row] = consumer.poll(pollTimeout).asScala.map(_.key.get)
+            rows.foreach(r => blockingQueue.add(r))
+            log.trace("  queue size: {}, row size: {}", blockingQueue.size, rows.size)
           }
         } else {
-          log.trace("  paused. queue size: {}", queue.size)
+          log.trace("  paused. queue size: {}", blockingQueue.size)
         }
-        Thread.sleep(50)
+        Thread.sleep(10)
```
Comment (chia7712) [inline ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 為何這裡需要sleep?

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala [hunk 2]
```diff
@@ -96,6 +82,8 @@ private[shabondi] class RowQueueProducer(
     stop()
   }
 
+  def queue: BlockingQueue[Row] = blockingQueue
```
Comment (chia7712) [inline ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 暴露這個方法會導致"外界"有機會去修改物件內部狀態，如果非要暴露的話，要盡量限縮使用範圍。

例如這邊的暴露應該是為了測試，那麼就要加上private[scope]去限縮

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala (hunk 1 above)
Comment (vitojeng) [reply ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 是為了避免 thread 空轉

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala [hunk 3]
```diff
@@ -14,53 +14,39 @@
...
 private[shabondi] object RowQueueProducer {
-  def apply(
-    queue: BlockingQueue[Row],
-    brokerProps: String,
-    topicNames: Seq[String],
-    pollTimeout: FiniteDuration,
-    pollSize: Int
-  ) =
-    new RowQueueProducer(queue, brokerProps, topicNames, pollTimeout, pollSize)
+  def apply(brokerProps: String, topicNames: Seq[String], pollTimeout: FiniteDuration) =
```
Comment (chia7712) [inline ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 看起來"正統的"建構方式應該是利用Config，而這邊這個apply看起來應該是給測試用的(?)

如果是的話，這個建構式應該放到test去，少一點建構式可以讓我們更明確知道一個物件應該"如何"被建立

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala (hunk 2 above)
Comment (vitojeng) [reply ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 目前己經在 class 加上 private[shabondi], 這樣是否 ok ?

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala (hunk 3 above)
Comment (vitojeng) [reply ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: > 而這邊這個apply看起來應該是給測試用的(?)

是的, 等一下調整.

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala (hunk 1 above)
Comment (vitojeng) [reply ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 看起來調整一下 sleep 的位置比較好...

Comment (chia7712) [reply ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 應該放在pause=true的時候

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala (hunk 2 above)
Comment (chia7712) [reply ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: nice

File: ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala (hunk 1 above)
Comment (vitojeng) [reply ohara-shabondi/src/main/scala/com/island/ohara/shabondi/sink/RowQueueProducer.scala]: 感謝~

**Comment by vitojeng** [comment]
retry

**Comment by vitojeng** [comment]
unrelated error

**Pull request merged**